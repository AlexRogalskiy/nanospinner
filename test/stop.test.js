let { createSpinner } = require('../index.js')

let stdout = { out: '' }
stdout.write = (symbols) => {
  stdout.out += symbols
}

it(`doesn't reprint stop message`, () => {
  stdout.out = ''
  let spinner = createSpinner('#stop', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.stop()

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})

it('stops after 2 spins and prints stop message', () => {
  stdout.out = ''
  let spinner = createSpinner('#stop', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.stop({ text: 'Done!' })

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})

it('marks spinner as stop with mark', () => {
  stdout.out = ''
  let spinner = createSpinner('#stop', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.stop({ mark: 'V' })

  expect(stdout.out).toMatchSnapshot(process.env.CI ? 'CI' : 'Local')
})
