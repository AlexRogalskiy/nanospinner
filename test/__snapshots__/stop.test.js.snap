// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't reprint stop message: CI 1`] = `
"[33m-[39m #stop
[33m-[39m #stop
[33m-[39m #stop
[33m-[39m #stop
"
`;

exports[`doesn't reprint stop message: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #stop[?25l[1G[2K[1G[33mâ ™[39m #stop[?25l[1G[2K[1G[33mâ ¹[39m #stop[1G[2K[1G[33mâ ¸[39m #stop
[?25h"
`;

exports[`marks spinner as stop with mark: CI 1`] = `
"[33m-[39m #stop
[33m-[39m #stop
[33m-[39m #stop
V #stop
"
`;

exports[`marks spinner as stop with mark: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #stop[?25l[1G[2K[1G[33mâ ™[39m #stop[?25l[1G[2K[1G[33mâ ¹[39m #stop[1G[2K[1GV #stop
[?25h"
`;

exports[`stops after 2 spins and prints stop message: CI 1`] = `
"[33m-[39m #stop
[33m-[39m #stop
[33m-[39m Done!
"
`;

exports[`stops after 2 spins and prints stop message: Local 1`] = `
"[?25l[1G[33mâ ‹[39m #stop[?25l[1G[2K[1G[33mâ ™[39m #stop[1G[2K[1G[33mâ ¹[39m Done!
[?25h"
`;
